src/DETHRACE/common/car.c

```diff
@@ -641,53 +631,53 @@ void GetFacesInBox(tCollision_info* c) {
         BrMatrix34Mul(&mat6, &mat5, &mat4);
         BrMatrix34LPInverse(&mat5, &mat6);
         GetNewBoundingBox(&predicted_bounds, c->bounds, &mat5);
-        for (i = 0; i < 3; ++i) {
+        for (i = 0; i < 3; i++) {
             if (bnds.original_bounds.min.v[i] > predicted_bounds.min.v[i]) {
                 bnds.original_bounds.min.v[i] = predicted_bounds.min.v[i];
             }
             if (bnds.original_bounds.max.v[i] < predicted_bounds.max.v[i]) {
                 bnds.original_bounds.max.v[i] = predicted_bounds.max.v[i];
             }
-            bnds.original_bounds.min.v[i] = bnds.original_bounds.min.v[i] - 0.02;
-            bnds.original_bounds.max.v[i] = bnds.original_bounds.max.v[i] + 0.02;
+            bnds.original_bounds.min.v[i] -= 0.02f;
+            bnds.original_bounds.max.v[i] += 0.02f;
         }
         c->last_box = bnds.original_bounds;
         BrMatrix34Copy(&c->last_box_inv_mat, &mat3);
         bnds.mat = &mat;
         c->box_face_start = gFace_count;
         gPling_face = NULL;
-        gFace_count += FindFacesInBox(&bnds, &gFace_list__car[gFace_count], 150 - gFace_count);
-        if (gFace_count >= 150) {
+        gFace_count += FindFacesInBox(&bnds, &gFace_list__car[gFace_count], COUNT_OF(gFace_list__car) - gFace_count);
+        if (gFace_count >= COUNT_OF(gFace_list__car)) {
             c->box_face_start = 0;
-            gFace_count = FindFacesInBox(&bnds, gFace_list__car, 150);
+            gFace_count = FindFacesInBox(&bnds, gFace_list__car, COUNT_OF(gFace_list__car));
             ++gFace_num__car;
         }
         old_d = c->water_d;
         if (c->driver == eDriver_local_human
-            && c->water_d != 10000.0
+            && c->water_d != 10000.f
             && gDouble_pling_water
-            && c->bounds_world_space.max.v[2] * c->water_normal.v[2] + c->bounds_world_space.max.v[1] * c->water_normal.v[1] + c->water_normal.v[0] * c->bounds_world_space.max.v[0] - c->water_d <= 0.0) {
+            && BrVector3Dot(&c->bounds_world_space.max, &c->water_normal) - c->water_d <= 0.f) {
             gInTheSea = 1;
             FreezeCamera();
         }
-        if (gPling_face && fabs(gPling_face->normal.v[1]) > 0.89999998) {
-            c->water_normal = gPling_face->normal;
-            if (c->water_normal.v[1] < 0.0) {
+        if (gPling_face != NULL && fabsf(gPling_face->normal.v[1]) > 0.9f) {
```

Comment:
Thinking about this, this might be `>=` because the `0.89999998` is 1 eps smaller then 0.9.
This is probably a compiler optimization that I've been interpreting wrong thus far.
```suggestion
        if (gPling_face != NULL && fabsf(gPling_face->normal.v[1]) >= 0.9f) {
```

Reply:
Or this is just a matter of floats not being able to represent a decimal number exactly.

Reply:
Agreed. I've been gradually dropping those extra decimals where I've been touching the code. Early on I didn't have that realization and copied them as-is into the code base

Reply:
I mean, just dropping the decimal points, not changing the comparision